### PR TITLE
Feat/add snowflake plots package

### LINKED ISSUE #81: Refactor plotting into SnowflakePlots

### PR DESCRIPTION
**Pull request by npoirie1**
Moves the visualization tools to a new package called SnowflakePlots.jl.

Fixes #81

### CODE REVIEW

**Review by plerouxanyon — COMMENTED** [review]
Thanks!

I think we should remove test files since they don't test anything. We can come back later and refactor for testability.

It also needs to be updated with the latest changes in `main`.

Other than that, it looks good!

**Comment by npoirie1** [comment]
@plerouxanyon, SnowflakePlots has been updated based on the latest changes in `main`. I have also removed the tests that were in the SnowflakePlots package. Do you think that it's a good idea to have separate code coverage reports for Snowflake and SnowflakePlots?

**Comment by plerouxanyon** [comment]
> @plerouxanyon, SnowflakePlots has been updated based on the latest changes in `main`. I have also removed the tests that were in the SnowflakePlots package. Do you think that it's a good idea to have separate code coverage reports for Snowflake and SnowflakePlots?

Thanks!

Regarding the coverage: If it can be easily split, then I think that would be good, but otherwise, we can keep them together for now.

**Comment by npoirie1** [comment]
> 

Splitting the coverage won't be easy. From what I can tell, it requires adding SnowflakePlots to Codecov. We should keep the coverage as is for now.

**Review by plerouxanyon — CHANGES_REQUESTED** [review]
Just the dependencies that need to be fixed, then we can merge.

File: SnowflakePlots.jl/Project.toml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+name = "SnowflakePlots"
+uuid = "fa4a7b36-de81-424e-b6b4-6a78b5e46525"
+authors = ["Alireza Yazdi <alireza@anyonsys.com>"]
+version = "0.1.0"
+
+[deps]
+Coverage = "a2441757-f6aa-5fb2-8edb-039e3f45d037"
```
[the remark this answers is not in the document]
Comment (npoirie1) [reply SnowflakePlots.jl/Project.toml:7]: The `Test` dependency has been removed. However, the `Coverage` dependency is needed for the `coverage.jl` file. As for the `Revise` package, it is useful when developing.

[the remark this answers is not in the document]
Comment (plerouxanyon) [reply SnowflakePlots.jl/Project.toml:7]: The `Coverage` package is a dependency of the tests and not of `SnowflakePlots`. See https://pkgdocs.julialang.org/v1/creating-packages/#Test-specific-dependencies-in-Julia-1.2-and-above.

As for `Revise`, I still think we should remove it. We don't want to force users to use a package that they don't necessarily want. A developer could always add it to their local environment if they want to use it.

[the remark this answers is not in the document]
Comment (npoirie1) [reply SnowflakePlots.jl/Project.toml]: `Coverage` and `Revise` are no longer dependencies.

**Review by plerouxanyon — APPROVED** [review]
Looks good!

**Pull request merged**